ietf/liaisons/tests.py

```diff
@@ -402,6 +404,13 @@ def test_edit_liaison(self):
         self.assertEqual(r.status_code, 200)
         q = PyQuery(r.content)
         self.assertEqual(len(q('form input[name=from_contact]')), 1)
+        json_data = q('form select[name=related_to]').attr('data-pre')
+        try:
+            decoded = json.loads(json_data)
+        except json.JSONDecodeError as e:
+            self.fail('data-pre contained invalid JSON data: %s' % str(e))
```

Comment:
This is a little odd. The test is "Does the code that builds the form put JSON in the field". I guess it's better to break of there's something that's not clearly JSON there, but a lot could go in that would survive `json.loads` that isn't useful.

`self.fail` outside of something like `setUp` where you are using it to check that the test framework and factory/fixture data is sane is always a bit of a red flag.

A better test would look to see that the fieid contained the expected data?

Reply:
That pattern exists elsewhere in the tests, so we should fix it everywhere.